### DISCUSSION TITLE
How to run using a mechanism with arbitrary reaction order

### DISCUSSION THREAD

**Discussion by JayStandridge** [AMReX-Combustion/PeleC#699] [Q&A] [answered]
PelePhysics has the Westbrook and Dryer two-step mechanism as well as the Jones and Lindstedt four-step mechanism, these global mechanisms have non-stoichiometric reaction orders and seem to be unsupported. I would like to study these mechanisms, is there a way to enable them, or get the convert.sh to work for these mechanisms?

Thanks in advance

**Comment by marchdf** [comment]
@malihass has looked into this and may be able to comment better than I can.

**Comment by malihass** [comment, marked as answer]
Thanks for the question!
There was a recent update in PelePhysics (https://github.com/AMReX-Combustion/PelePhysics/pull/414) that enabled the use of arbitrary order mechanisms `chem-CH4-2step` and `JL4` (which you referred to) including with analytical chemical Jacobian. The `list_mech` file was also updated to reflect this change. Note that I tested these mechs in 0D only.
If you update your PelePhysics submodule, you should be able to run these mechs.
